bindings/node.js/kzg.cxx

```diff
@@ -0,0 +1,341 @@
+    return env.Null();
+  }
+
+  auto c_param = info[0].As<Napi::TypedArray>();
+  if (c_param.TypedArrayType() != napi_uint8_array) {
+    Napi::Error::New(env, "Expected an Uint8Array")
+        .ThrowAsJavaScriptException();
+    return env.Undefined();
+  }
+  auto polynomialKzg = c_param.As<Napi::Uint8Array>().Data();
+
+  auto z_param = info[0].As<Napi::TypedArray>();
+  if (z_param.TypedArrayType() != napi_uint8_array) {
+    Napi::Error::New(env, "Expected an Uint8Array")
+        .ThrowAsJavaScriptException();
+    return env.Undefined();
+  }
+  auto z = z_param.As<Napi::Uint8Array>().Data();
+
+  auto y_param = info[0].As<Napi::TypedArray>();
+  if (y_param.TypedArrayType() != napi_uint8_array) {
+    Napi::Error::New(env, "Expected an Uint8Array")
+        .ThrowAsJavaScriptException();
+    return env.Undefined();
+  }
+  auto y = y_param.As<Napi::Uint8Array>().Data();
+
+  auto proof_param = info[0].As<Napi::TypedArray>();
+  if (proof_param.TypedArrayType() != napi_uint8_array) {
+    Napi::Error::New(info.Env(), "Expected an Uint8Array")
+        .ThrowAsJavaScriptException();
+    return info.Env().Undefined();
+  }
+  auto kzgProof = proof_param.As<Napi::Uint8Array>().Data();
+
+  auto kzgSettings = info[4].As<Napi::External<KZGSettings>>().Data();
+
+  BLSFieldElement fz, fy;
+  bytes_to_bls_field(&fz, z);
+  bytes_to_bls_field(&fy, y);
+
+  KZGCommitment commitment;
+  auto ret = bytes_to_g1(&commitment, polynomialKzg);
+  if (ret != C_KZG_OK) {
+    std::ostringstream ss;
+    std::copy(polynomialKzg, polynomialKzg + BYTES_PER_COMMITMENT, std::ostream_iterator<int>(ss, ","));
+
+    Napi::TypeError::New(env, "Failed to parse argument commitment: "  + ss.str() + " Return code was: " + std::to_string(ret)).ThrowAsJavaScriptException();
+    return env.Null();
+  };
+
+  KZGProof proof;
+  if (bytes_to_g1(&proof, kzgProof) != C_KZG_OK) {
+    Napi::TypeError::New(env, "Invalid kzgProof").ThrowAsJavaScriptException();
+    return env.Null();
+  }
+
+  bool out;
+  if (verify_kzg_proof(&out, &commitment, &fz, &fy, &proof, kzgSettings) == C_KZG_OK) {
+    return Napi::Boolean::New(env, true);
+  }
+
+  return Napi::Boolean::New(env, false);
+}
+
+Napi::Object Init(Napi::Env env, Napi::Object exports) {
+  // Functions
+  exports["loadTrustedSetup"] = Napi::Function::New(env, LoadTrustedSetup);
+  exports["freeTrustedSetup"] = Napi::Function::New(env, FreeTrustedSetup);
+  exports["verifyKzgProof"] = Napi::Function::New(env, VerifyKzgProof);
+  exports["blobToKzgCommitment"] = Napi::Function::New(env, BlobToKzgCommitment);
+  exports["computeAggregateKzgProof"] = Napi::Function::New(env, ComputeAggregateKzgProof);
+  exports["verifyAggregateKzgProof"] = Napi::Function::New(env, VerifyAggregateKzgProof);
+
+  // Constants
+  exports["FIELD_ELEMENTS_PER_BLOB"] = Napi::Number::New(env, FIELD_ELEMENTS_PER_BLOB);
+  exports["BYTES_PER_FIELD"] = Napi::Number::New(env, BYTES_PER_FIELD);
+
+  return exports;
```

Comment:
This is the interface of the JavaScript module which can be imported by a NodeJS program.